sled-agent/src/dump_setup.rs

```diff
@@ -1129,13 +1134,49 @@ impl DumpSetupWorker {
         for path in
             glob::glob(debug_dir.as_ref().join("**/*").as_str())?.flatten()
         {
-            let meta = tokio::fs::metadata(&path).await?;
+            let meta = match tokio::fs::metadata(&path).await {
+                Ok(meta) => meta,
+                // "Not found" errors could be caused by TOCTTOU -- observing a
+                // file which gets renamed, removed, etc. We'll ignore these
+                // files.
+                Err(err)
```

Comment:
I didn't see this TOCTTOU issue explicitly, but it seems possible, and it would prevent cleanup.

Reply:
(One example case where it's possible - the support bundles make a temporary file which we see when we "glob", and which gets renamed when we try to query "metadata")